### ISSUE TITLE
Add option to not use namespaces

### ISSUE THREAD

**Issue by a-churchill** [thesayyn/protoc-gen-ts#79]
The generated code uses namespaces to organize the generated members; however, `create-react-app`'s built-in Babel config does not support namespaces. The error message:

```
SyntaxError: Namespace not marked type-only declare. Non-declarative namespaces are only supported experimentally in Babel. To enable and review caveats see: https://babeljs.io/docs/en/babel-plugin-transform-typescript
```

Enabling namespaces is not possible with `create-react-app`. Is it possible to add an option to the generator to not use namespaces?

**Comment by thesayyn** [comment]
It would be possible but it’d require a lot of refactoring.

**Comment by thesayyn** [comment]
Also, upon my work on public import, I came to realize that handling this will be so hard. Not impossible but would require a lot of effort.

**Comment by ImperiumTakp** [comment]
i also encountered this issue,

**Comment by relief-melone** [comment]
I would also like this. Another option which i would like even better (as protobufs just rely on namespaces alot even though its not the most favoured concept in TS) is the output of a single file because otherwise namespaces tend to conflict alot with each other. Would also contribute. Having trouble at the moment to get the debugger running :D. Also one addition. Is there a reason why this project is not written in TypeScript?

**Comment by efossvold** [comment]
This would be much as appreciated as we really need this for our project as well.

**Comment by thesayyn** [comment]
> Having trouble at the moment to get the debugger running :D. Also one addition. Is there a reason why this project is not written in TypeScript?

When I started writing the plugin it was small and I didn’t want to deal with all that transpiling stuff.

**Comment by thesayyn** [comment]
> This would be much as appreciated as we really need this for our project as well.

I am willing to accept a decent PR that handles name conflicts and other possible hurdles along the way. there is #98 but it is not complete yet.  

I can’t work on this myself atm as I am fed up with the tasks of my full-time job.

**Comment by chris-kruining** [comment]
I managed to implement this in #112

**Comment by chris-kruining** [comment]
has become #117

**Comment by thesayyn** [comment]
fixed by #132
